provider/cluster_rosa_classic_resource.go

```diff
@@ -645,20 +670,40 @@ func createClassicClusterObject(ctx context.Context,
 	return object, err
 }
 
-func (r *ClusterRosaClassicResource) validateAccountRoles(ctx context.Context, state *ClusterRosaClassicState) error {
-	r.logger.Debug(ctx, "Validating if cluster version is compatible to account roles' version")
-	region := state.CloudRegion.Value
-	version := ""
+// getAndValidateVersion ensures that the cluster version is available in the
+// channel group
+func (r *ClusterRosaClassicResource) getAndValidateVersion(ctx context.Context, state *ClusterRosaClassicState) (string, error) {
+	channelGroup := ocm.DefaultChannelGroup
+	if !state.ChannelGroup.Unknown && !state.ChannelGroup.Null {
+		channelGroup = state.ChannelGroup.Value
+	}
+
+	versionList, err := r.getVersionList(r.logger, ctx, channelGroup)
+	if err != nil {
+		return "", err
+	}
+
+	version := versionList[0]
 	if !state.Version.Unknown && !state.Version.Null {
-		version = state.Version.Value
+		version = strings.Replace(state.Version.Value, "openshift-v", "", 1)
 	}
 
-	if version == "" {
-		versionList, err := r.getVersionList(r.logger, ctx)
-		if err != nil {
-			return err
+	r.logger.Debug(ctx, "Validating if cluster version %s is in the list of supported versions: %v", version, versionList)
+	for _, v := range versionList {
+		if v == version {
+			return version, nil
 		}
-		version = versionList[0]
+	}
+
+	return "", fmt.Errorf("version %s is not in the list of supported versions: %v", version, versionList)
+}
+
+func (r *ClusterRosaClassicResource) validateAccountRoles(ctx context.Context, state *ClusterRosaClassicState) error {
+	r.logger.Debug(ctx, "Validating if cluster version is compatible to account roles' version")
+	region := state.CloudRegion.Value
+	version, err := r.getAndValidateVersion(ctx, state)
```

Comment:
I'm not particularly pleased at putting the version validation inside the account role validation function, but this code already made the calls to retrieve the version list, so throwing the check in here was the easiest place.
As an alternative, I could update things to memoize the version list and check it separately.